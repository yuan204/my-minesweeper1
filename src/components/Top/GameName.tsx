import styled from "@emotion/styled";
export interface GameNameProps {
  children: string;
}
export const GameName = styled.h1<GameNameProps>`
  font-size: 2em;
`;
