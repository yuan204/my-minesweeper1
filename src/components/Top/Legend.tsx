import styled from "@emotion/styled";
import React, { FC } from "react";

export interface LegendProps {
  feature: string;
  firstAction: string;
  secondAction: string;
}

export const Legend: FC<LegendProps> = ({
  feature,
  firstAction,
  secondAction,
}) => {
  return (
    <LegendContainer>
      <strong>{feature}: </strong>
      <Action>
        <FirstAction>{firstAction}</FirstAction>+
        <SecondAction>{secondAction}</SecondAction>
      </Action>
    </LegendContainer>
  );
};

const LegendContainer = styled.strong`
  font-size: 1em;
  margin: 0 auto 2vw;
  line-height: 1.25em;
`;

const Action = styled.code`
  background: #e3e3e3;
`;

const FirstAction = styled.code`
  color: #ec433c;
`;

const SecondAction = styled.code`
  color: #2a48ec;
`;
