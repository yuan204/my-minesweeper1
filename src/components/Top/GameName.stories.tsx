import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { GameName } from "./GameName";

export default {
  title: "Top/GameName",
  component: GameName,
} as ComponentMeta<typeof GameName>;

const Template: ComponentStory<typeof GameName> = (args) => (
  <GameName {...args} />
);

export const GameNameStory = Template.bind({});
GameNameStory.args = {
  children: "Minesweeper",
};
