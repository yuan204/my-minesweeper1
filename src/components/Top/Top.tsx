import { Legend, LegendProps } from "./Legend";
import { GameName, GameNameProps } from "./GameName";
import styled from "@emotion/styled";
import React, { FC } from "react";

type TopProps = LegendProps & GameNameProps;
export const Top: FC<TopProps> = ({ children, ...legendProps }) => {
  return (
    <Header>
      <GameName>{children}</GameName>
      <Legend {...legendProps} />
    </Header>
  );
};

const Header = styled.header`
  text-align: center;
  position: relative;
  display: inline-block;
`;
