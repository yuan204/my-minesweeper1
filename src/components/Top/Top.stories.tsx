import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { Top } from "./Top";

export default {
  title: "Top/Top",
  component: Top,
} as ComponentMeta<typeof Top>;

const Template: ComponentStory<typeof Top> = (args) => <Top {...args} />;

export const TopPanel = Template.bind({});
TopPanel.args = {
  feature: "Flag",
  firstAction: "ctrl",
  secondAction: "click",
  children: "Minesweeper",
};
