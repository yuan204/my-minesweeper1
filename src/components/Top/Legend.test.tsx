import { render } from "@testing-library/react";
import { GameName } from "@/components/Top/GameName";
import React from "react";
import { Legend } from "@/components/Top/Legend";

test("legend should render correctly", () => {
  const props = {
    feature: "Flag",
    firstAction: "ctrl",
    secondAction: "click",
  };
  const { asFragment } = render(<Legend {...props} />);
  expect(asFragment()).toMatchSnapshot();
});
