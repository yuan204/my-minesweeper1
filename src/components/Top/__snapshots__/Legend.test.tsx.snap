// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`legend should render correctly 1`] = `
<DocumentFragment>
  <strong
    class="css-17qz6a2"
  >
    <strong>
      Flag: 
    </strong>
    <code
      class="css-12ap3z0"
    >
      <code
        class="css-udfepp"
      >
        ctrl
      </code>
      +
      <code
        class="css-9c23k1"
      >
        click
      </code>
    </code>
  </strong>
</DocumentFragment>
`;
